code/jobs/DeployJob.php

```diff
@@ -80,6 +80,22 @@ public function perform() {
 		$deployment = DNDeployment::get()->byID($this->args['deploymentID']);
 		$environment = $deployment->Environment();
 		$project = $environment->Project();
+		$backupDataTransfer = null;
+		$backupMode = !empty($this->args['backup_mode']) ? $this->args['backup_mode'] : 'db';
+
+		if(!empty($this->args['backup_before_deploy'])) {
```

Comment:
I have a suggestion: how about this option is controlled on three levels:
1) backend-specific default ("on" for Capistrano, "off" for Rainforest)
2) overriden by CMS per-environment setting 
3) overriden by user in "advanced options"

We can then skip the implementation of (3) because the UI is changing anyway and maybe people don't even need it - we could try shipping it and adding it later if someone complains :-)